driver/src/main/scala/api/database.scala

```diff
@@ -140,13 +144,14 @@ trait DBMetaCommands {
 
   /** Returns the names of the collections in this database. */
   def collectionNames(implicit ec: ExecutionContext): Future[List[String]] = {
-    // TODO: Use command { 'listCollections': 1 }, check with WT
-    collection("system.namespaces").as[BSONCollection]()
+    val wireVer = connection.metadata.map(_.maxWireVersion)
+
+    if (wireVer.exists(_ == MongoWireVersion.V30)) {
+      Command.run(BSONSerializationPack)(self, ListCollectionNames).map(_.names)
+    } else collection("system.namespaces").as[BSONCollection]()
       .find(BSONDocument(
         "name" -> BSONRegex("^[^\\$]+$", "") // strip off any indexes
```

Comment:
The tests about that on MongoDB < 3 are (still) passing with it. Have you seen any issue about?